Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,443 @@
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.git_status.ref | Should -Be $pullRequest.head.ref
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings from a PullRequest' {
+            BeforeAll {
+                  $repoWithPR = Get-GitHubRepository -OrganizationName $script:organizationName |
+                      Where-Object { $_ | Get-GitHubPullRequest } |
+                      Select-Object -First 1
+                $pullRequest = $repoWithPR | Get-GitHubPullRequest | Select-Object -First 1
+                $newGitHubCodespaceParms = @{
+                    PullRequest = $pullRequest.number
+                    RepositoryId = $repoWithPR.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repoWithPR.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.pulls_url | Should -Be $pullRequest.url
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with all possible settings' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    # ClientIp = 'TODO ???? - should be instead of rather than in addition to Geo, perhaps add some param validation to the function'
+                    # DevContainerPath = 'Will add to test in the future when Get-GitHubDevContainer is implemented and the test repo includes one'
+                    DisplayName = 'PowerShellForGitHub pester test'
+                    Geo = 'UsWest'
+                    Machine = 'basicLinux32gb'
```

Comment:
I was looking for where these names came from.  Eventually found [this](https://docs.github.com/en/rest/codespaces/machines?apiVersion=2022-11-28):

```powershell
(Invoke-GHRestMethod -UriFragment '/repos/PowerShellForGitHubTeam/6cd3a09f-9275-45a6-b908-f0e6b294afcf/codespaces/machines' -Method Get -Description 'Get available Codespace machine types').machines
```

We should definitely get your current change checked-in (once these remaining review comments are resolved), but a TODO in my head is to get those two API's added as well so that users can find out what possible machines can be used when using this module.

Reply:
Reasonable. Will try to remember to add both [machines](https://docs.github.com/en/rest/codespaces/machines?apiVersion=2022-11-28) endpoints in the next PR.  These would be good candidates for ArgumentCompleters, but I see you don't have any - probably due to compatibility concerns.

Reply:
I'm referring to https://learn.microsoft.com/en-us/previous-versions/powershell/module/microsoft.powershell.core/about/about_functions_argument_completion?view=powershell-7.1.  I think it was officially introduced in PowerShell 5, but I also think there is a module which provides Register-ArgumentCompleter support for older versions.

Reply:
Ah, thanks.  We were looking at the same docs, just different ways of accomplishing it.  The `IValidateSetValuesGenerator` approach requires `Classes` which weren't introduced until PS6.  It looks like the `ArgumentCompletions` attribute was made available in PS5.  I'm not quite sure how I feel about having to do a network call for tab completion to work.  While nifty, I think that approach is likely overkill for our immediate needs, since an invalid value will likely get a reasonable error message from the API.  I think it should be enough for there to be an available API for uses to call to know what possible values could be used for that.  But again, this is something that can be hashed out further in a future PR.

Reply:
Agreed.